ecr-login/cache/file.go

```diff
@@ -98,6 +98,9 @@ func (f *fileCredentialCache) fullFilePath() string {
 // file access. There is not guarantee here for handling multiple writes at once since there is no out of process locking.
 func (f *fileCredentialCache) save(registryCache *RegistryCache) error {
 	defer log.Flush()
+	if _, err := os.Stat(f.path); err != nil {
```

Comment:
The takeaway is we won't know what is the directory configuration is until we make the call. The log config is initialized before the call is made, in result, the log and the cache will end up in different directory which can also be confusing.